### PR TITLE
tests: add common tag prefix for cloud instances and groups

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message
```
To aid in debug of test instances launched by cloud-init, use
a common tag prefix format:
   ci-<platform>-<release>-MMDD-<timestamp>[0-9a-z]{8}

Example instance names:
   ci-lxd-container-jammy-0304-144058la5mnhu9
```

## Additional Context
SC-840

## Test Steps
```
CLOUD_INIT_PLATFORM=lxd_container CLOUD_INIT_KEEP_INSTANCE=true CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/daily CLOUD_INIT_OS_IMAGE=jammy::ubuntu::jammy .tox/integration-tests/bin/pytest tests/integration_tests/modules/test_apt.py::TestApt::test_ppa_source

$ lxc ls ci
 lxc ls ci
+--------------------------------------------+---------+--------------------+-----------------------------------------------+-----------+-----------+
|                    NAME                    |  STATE  |        IPV4        |                     IPV6                      |   TYPE    | SNAPSHOTS |
+--------------------------------------------+---------+--------------------+-----------------------------------------------+-----------+-----------+
| ci-lxd-container-jammy-0304-144058la5mnhu9 | RUNNING | 10.69.12.43 (eth0) | fd42:fc73:a2ea:bdfb:216:3eff:feb7:47ca (eth0) | CONTAINER | 0         |
+--------------------------------------------+---------+--------------------+-----------------------------------------------+-----------+-----------+
```

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by holmanb — APPROVED** [review]
Looks good to me. Tested locally and verified lxd container name is as expected. Approved, pending the failing CI tests.

File: tests/integration_tests/clouds.py
Comment (paride) [inline tests/integration_tests/clouds.py:95]: This fails for me (Python 3.10):
```
tests/integration_tests/conftest.py:16: in <module>
    from tests.integration_tests.clouds import (
tests/integration_tests/clouds.py:91: in <module>
    class IntegrationCloud(ABC):
tests/integration_tests/clouds.py:95: in IntegrationCloud
    pycloudlib_instance_cls = Type(BaseCloud)
/usr/lib/python3.10/typing.py:941: in __call__
    raise TypeError(f"Type {self._name} cannot be instantiated; "
E   TypeError: Type Type cannot be instantiated; use type() instead
```
Replacing with `type()` works.

File: tests/integration_tests/clouds.py
Comment (paride) [inline tests/integration_tests/clouds.py:94]: This looks unused?

File: tests/integration_tests/clouds.py
Comment (paride) [inline tests/integration_tests/clouds.py:293]: Maybe in pycloudlib we should make LXD use the `tag` to set the instances default name, like in the other clouds. (Not to be fixed here.)

File: tests/integration_tests/clouds.py
Comment (paride) [inline tests/integration_tests/clouds.py:294]: Now we have to "moving parts" in the instance naming: the timestamp set in pycloudlib and this random string. In Jenkins we may want to use fixed instance names to prevent the proliferation of stale instances should we have bugs in the cleanup code. This would require disabling both this random string and the pycloudlib timestamp.

I was thinking of dropping this random string bit, leaving it to pycloudlib to create unique instance names based on the tag. Then we can add a `pycloudlib.toml` setting to make it not add the timestamp in the cases where we want fixed instance names instead.

We already have seconds resolution in the pycloudlib timestamping, hopefully that's enough to prevent collisions. Thoughts?

**Review by paride — CHANGES_REQUESTED** [review]
Big improvement in handling the instance naming, thanks! I left a few inline comments, where the more important one is a RFC about dropping the random part of the instance names.

File: tests/integration_tests/clouds.py
Comment (paride) [reply tests/integration_tests/clouds.py:121]: In any case black wants double quotes here.

Comment (blackboxsw) [reply tests/integration_tests/clouds.py:121]: +1 on that collapse is good.

File: tests/integration_tests/clouds.py
Comment (blackboxsw) [reply tests/integration_tests/clouds.py]: I had a typo/thinko
```suggestion
    pycloudlib_instance_cls: Type[BaseCloud]
```

File: tests/integration_tests/clouds.py
Comment (blackboxsw) [reply tests/integration_tests/clouds.py:293]: I think that makes sense in pycloudlib so all clouds behave the same-ish,

File: tests/integration_tests/clouds.py
Comment (blackboxsw) [reply tests/integration_tests/clouds.py:294]: Yes this makes sense. If we have parallel jobs running in jenkins, it's likely they'll be different Ubuntu series or platforms so collisions should not generally occur. If we end up runing jobs for a single platform-series in parallel instead of sequentially at some point we can grow random chars at that point if we actually hit a collision problem. I'll drop this from this branch and we can incorporate that into pycloudlib in the future if needed.

Comment (paride) [reply tests/integration_tests/clouds.py:294]: +1 on incorporating this into pycloudlib if needed.

Comment (blackboxsw) [reply tests/integration_tests/clouds.py:294]: Had to delete my prior comment on this. I agree with the sentiement about pycloudlib randomizing instance name based on the cloud.tag prefix. The issue we have at the moment in cloud-init CI is that pycloudlib doesn't yet have this feature, and the tag is created only at the start of integration tests. So, until pycloudlib grows this feature to create names based on tags, this branch should either go to WIP/draft and await our work on pycloudlib, or add this random string to allow sequentially running mulitple integration tests in sequence. I'm good with WIP/draft on this PR if we want to add the pycloudlib feature first.

**Review by paride — COMMENTED** [review]
+1 on landing this modulo the review comments, when they make sense :) Go ahead without my explicit approval as I'll be on pto until Monday.

**Comment by blackboxsw** [comment]
Marked draft as I didn't get around to putting up the pycloudlib prerequisite PR so that we can drop the rand suffix up in cloud-init integration tests. Will get back to this in a week.

**Pull request closed without merging**
